src/driver/experimental_trial.py

```diff
@@ -212,6 +212,35 @@ def evaluate(self, seed: int) -> pd.DataFrame:
             ],
             axis=1,
         )
+        if self._trial_descriptor.modeling_strategy == "m3strategy":
```

Comment:
Move this code to a function.  Also, is there something specific to the m3strategy in this code?  If not, remove the check for m3strategy.

Reply:
There's something specific to the m3strategy, which unfortunately makes the codes ugly.  Here I use `self.reach_surface._reach_curves`, which is an attribute in multi-pub models but not in other ReachSurfaces.  Specifically, not in single pub models nor in the `FakeSurface` in the unit test. 

Sure, will try to move this into a function and add unit test.